{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Estimate PRS Score for AFR and AMR"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Estimate PRS Score for GLGC AFR and AMR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/HDL_AFR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AFR\n",
      "  --extract ./data/snplist_data/AFR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/HDL_AFR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/HDL_AFR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 13:44:51 2025\n",
      "15000 MiB RAM detected, ~8058 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "53944 samples (0 females, 0 males, 53944 ambiguous; 53944 founders) loaded from\n",
      "./data/geno_data/AFR.fam.\n",
      "824452 variants loaded from ./data/geno_data/AFR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 824452 variants remaining.\n",
      "Calculating allele frequencies... 1523313947556371798795done.\n",
      "824452 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 3321 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/HDL_AFR_MIXPRS.txt were skipped due to\n",
      "missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 824406 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/HDL_AFR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 13:48:51 2025\n",
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/HDL_AMR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AMR\n",
      "  --extract ./data/snplist_data/AMR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/HDL_AMR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/HDL_AMR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 13:48:51 2025\n",
      "15000 MiB RAM detected, ~8113 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "40838 samples (0 females, 0 males, 40838 ambiguous; 40838 founders) loaded from\n",
      "./data/geno_data/AMR.fam.\n",
      "1105425 variants loaded from ./data/geno_data/AMR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 1105425 variants remaining.\n",
      "Calculating allele frequencies... 111723293541475359657177838894done.\n",
      "1105425 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 8653 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/HDL_AMR_MIXPRS.txt were skipped due to\n",
      "missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 1099270 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/HDL_AMR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 13:55:21 2025\n",
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/LDL_AFR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AFR\n",
      "  --extract ./data/snplist_data/AFR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/LDL_AFR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/LDL_AFR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 13:55:21 2025\n",
      "15000 MiB RAM detected, ~8064 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "53944 samples (0 females, 0 males, 53944 ambiguous; 53944 founders) loaded from\n",
      "./data/geno_data/AFR.fam.\n",
      "824452 variants loaded from ./data/geno_data/AFR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 824452 variants remaining.\n",
      "Calculating allele frequencies... 1523313947556371798795done.\n",
      "824452 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 3321 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/LDL_AFR_MIXPRS.txt were skipped due to\n",
      "missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 824406 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/LDL_AFR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 14:14:15 2025\n",
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/LDL_AMR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AMR\n",
      "  --extract ./data/snplist_data/AMR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/LDL_AMR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/LDL_AMR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 14:14:15 2025\n",
      "15000 MiB RAM detected, ~7989 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "40838 samples (0 females, 0 males, 40838 ambiguous; 40838 founders) loaded from\n",
      "./data/geno_data/AMR.fam.\n",
      "1105425 variants loaded from ./data/geno_data/AMR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 1105425 variants remaining.\n",
      "Calculating allele frequencies... 111723293541475359657177838894done.\n",
      "1105425 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 8592 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/LDL_AMR_MIXPRS.txt were skipped due to\n",
      "missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 1095925 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/LDL_AMR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 14:34:35 2025\n",
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/TC_AFR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AFR\n",
      "  --extract ./data/snplist_data/AFR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/TC_AFR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/TC_AFR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 14:34:35 2025\n",
      "15000 MiB RAM detected, ~7919 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "53944 samples (0 females, 0 males, 53944 ambiguous; 53944 founders) loaded from\n",
      "./data/geno_data/AFR.fam.\n",
      "824452 variants loaded from ./data/geno_data/AFR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 824452 variants remaining.\n",
      "Calculating allele frequencies... 1523313947556371798795done.\n",
      "824452 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 3321 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/TC_AFR_MIXPRS.txt were skipped due to\n",
      "missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 824406 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/TC_AFR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 14:53:46 2025\n",
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/TC_AMR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AMR\n",
      "  --extract ./data/snplist_data/AMR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/TC_AMR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/TC_AMR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 14:53:46 2025\n",
      "15000 MiB RAM detected, ~7870 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "40838 samples (0 females, 0 males, 40838 ambiguous; 40838 founders) loaded from\n",
      "./data/geno_data/AMR.fam.\n",
      "1105425 variants loaded from ./data/geno_data/AMR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 1105425 variants remaining.\n",
      "Calculating allele frequencies... 111723293541475359657177838894done.\n",
      "1105425 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 8723 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/TC_AMR_MIXPRS.txt were skipped due to\n",
      "missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 1102343 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/TC_AMR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 15:14:07 2025\n",
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/logTG_AFR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AFR\n",
      "  --extract ./data/snplist_data/AFR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/logTG_AFR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/logTG_AFR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 15:14:07 2025\n",
      "15000 MiB RAM detected, ~7804 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "53944 samples (0 females, 0 males, 53944 ambiguous; 53944 founders) loaded from\n",
      "./data/geno_data/AFR.fam.\n",
      "824452 variants loaded from ./data/geno_data/AFR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 824452 variants remaining.\n",
      "Calculating allele frequencies... 1523313947556371798795done.\n",
      "824452 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 3321 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/logTG_AFR_MIXPRS.txt were skipped due to\n",
      "missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 824406 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/logTG_AFR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 15:33:39 2025\n",
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/logTG_AMR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AMR\n",
      "  --extract ./data/snplist_data/AMR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/logTG_AMR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/logTG_AMR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 15:33:39 2025\n",
      "15000 MiB RAM detected, ~7766 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "40838 samples (0 females, 0 males, 40838 ambiguous; 40838 founders) loaded from\n",
      "./data/geno_data/AMR.fam.\n",
      "1105425 variants loaded from ./data/geno_data/AMR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 1105425 variants remaining.\n",
      "Calculating allele frequencies... 111723293541475359657177838894done.\n",
      "1105425 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 8687 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/logTG_AMR_MIXPRS.txt were skipped due to\n",
      "missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 1100439 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/logTG_AMR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 15:53:50 2025\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in HDL LDL TC logTG; do\n",
    "\n",
    "for pop2 in AFR AMR; do\n",
    "\n",
    "## MIXPRS\n",
    "method=MIXPRS\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${pop2}_${method}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${pop2}_${method}.txt header-read \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${pop2}_${method}\n",
    "fi\n",
    "\n",
    "done\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "## JointPRS\n",
    "for trait in HDL LDL TC logTG; do\n",
    "for pop2 in AFR AMR; do\n",
    "\n",
    "for pop in EUR EAS AFR SAS AMR; do\n",
    "\n",
    "method=JointPRS\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/MIXPRS/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_beta_${pop}_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/JointPRS/${trait}_${method}_meta_EUR_EAS_AFR_SAS_AMR_beta_${pop}.txt \\\n",
    "    --out ./result/score/diff_cohort/MIXPRS/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_beta_${pop}_prs_${pop2}\n",
    "fi\n",
    "\n",
    "done\n",
    "done\n",
    "\n",
    "done\n",
    "\n",
    "## SDPRX\n",
    "for trait in HDL LDL TC logTG; do\n",
    "for pop2 in AFR AMR; do\n",
    "\n",
    "for pop in EAS AFR SAS AMR; do\n",
    "\n",
    "method=SDPRX\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/MIXPRS/${trait}_${method}_EUR_${pop}_beta_${pop}_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_${pop}_beta_${pop}.txt \\\n",
    "    --out ./result/score/diff_cohort/MIXPRS/${trait}_${method}_EUR_${pop}_beta_${pop}_prs_${pop2}\n",
    "fi\n",
    "\n",
    "done\n",
    "\n",
    "method=SDPRX\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/MIXPRS/${trait}_${method}_EUR_${pop2}_beta_EUR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_beta_EUR.txt \\\n",
    "    --out ./result/score/diff_cohort/MIXPRS/${trait}_${method}_EUR_${pop2}_beta_EUR_prs_${pop2}\n",
    "fi\n",
    "\n",
    "done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in HDL LDL TC logTG; do\n",
    "\n",
    "for pop2 in AFR AMR; do\n",
    "\n",
    "## JointPRS_meta\n",
    "method=JointPRS_meta\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/JointPRS/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/JointPRS/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/JointPRS/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_prs_${pop2}\n",
    "fi\n",
    "\n",
    "done\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in HDL LDL TC logTG; do\n",
    "\n",
    "for pop2 in AFR AMR; do\n",
    "\n",
    "## JointPRS_linear\n",
    "method=JointPRS_linear\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/JointPRS/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/JointPRS/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_beta_${pop2}.txt header-read \\\n",
    "    --score-col-nums 3 4 5 6 7 \\\n",
    "    --out ./result/score/diff_cohort/JointPRS/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_prs_${pop2}\n",
    "fi\n",
    "\n",
    "done\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in HDL LDL TC logTG; do\n",
    "\n",
    "for pop2 in AFR AMR; do\n",
    "\n",
    "## SDPRX\n",
    "method=SDPRX\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_prs_${pop2}\n",
    "fi\n",
    "\n",
    "done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in HDL LDL TC logTG; do\n",
    "\n",
    "for pop2 in AFR AMR; do\n",
    "\n",
    "## XPASS\n",
    "method=XPASS\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in HDL LDL TC logTG; do\n",
    "\n",
    "for pop2 in AFR AMR; do\n",
    "\n",
    "## PRScsx\n",
    "method=PRScsx\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_beta_${pop2}.txt header-read \\\n",
    "    --score-col-nums 3 4 5 6 7 \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in HDL LDL TC logTG; do\n",
    "\n",
    "for pop2 in AFR AMR; do\n",
    "\n",
    "## MUSSEL\n",
    "method=MUSSEL\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in HDL LDL TC logTG; do\n",
    "\n",
    "for pop2 in AFR AMR; do\n",
    "\n",
    "## PROSPER\n",
    "method=PROSPER\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_SAS_AMR_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in HDL LDL TC logTG; do\n",
    "\n",
    "for pop2 in AFR AMR; do\n",
    "\n",
    "## PROSPER\n",
    "method=PROSPER\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_update_EUR_EAS_AFR_SAS_AMR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_update_EUR_EAS_AFR_SAS_AMR_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_update_EUR_EAS_AFR_SAS_AMR_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in HDL LDL TC logTG; do\n",
    "\n",
    "for pop2 in AFR AMR; do\n",
    "\n",
    "## BridgePRS\n",
    "method=BridgePRS\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Estimate PRS Score for PAGE AFR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/Height_AFR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AFR\n",
      "  --extract ./data/snplist_data/AFR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/Height_AFR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/Height_AFR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 15:53:52 2025\n",
      "15000 MiB RAM detected, ~7721 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "53944 samples (0 females, 0 males, 53944 ambiguous; 53944 founders) loaded from\n",
      "./data/geno_data/AFR.fam.\n",
      "824452 variants loaded from ./data/geno_data/AFR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 824452 variants remaining.\n",
      "Calculating allele frequencies... 1523313947556371798795done.\n",
      "824452 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 3325 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/Height_AFR_MIXPRS.txt were skipped due\n",
      "to missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 824413 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/Height_AFR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 16:13:21 2025\n",
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/BMI_AFR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AFR\n",
      "  --extract ./data/snplist_data/AFR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/BMI_AFR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/BMI_AFR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 16:13:22 2025\n",
      "15000 MiB RAM detected, ~7693 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "53944 samples (0 females, 0 males, 53944 ambiguous; 53944 founders) loaded from\n",
      "./data/geno_data/AFR.fam.\n",
      "824452 variants loaded from ./data/geno_data/AFR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 824452 variants remaining.\n",
      "Calculating allele frequencies... 1523313947556371798795done.\n",
      "824452 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 3325 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/BMI_AFR_MIXPRS.txt were skipped due to\n",
      "missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 824413 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/BMI_AFR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 16:32:54 2025\n",
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/SBP_AFR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AFR\n",
      "  --extract ./data/snplist_data/AFR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/SBP_AFR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/SBP_AFR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 16:32:54 2025\n",
      "15000 MiB RAM detected, ~7678 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "53944 samples (0 females, 0 males, 53944 ambiguous; 53944 founders) loaded from\n",
      "./data/geno_data/AFR.fam.\n",
      "824452 variants loaded from ./data/geno_data/AFR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 824452 variants remaining.\n",
      "Calculating allele frequencies... 1523313947556371798795done.\n",
      "824452 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 3325 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/SBP_AFR_MIXPRS.txt were skipped due to\n",
      "missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 824413 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/SBP_AFR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 16:52:27 2025\n",
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/DBP_AFR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AFR\n",
      "  --extract ./data/snplist_data/AFR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/DBP_AFR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/DBP_AFR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 16:52:27 2025\n",
      "15000 MiB RAM detected, ~7664 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "53944 samples (0 females, 0 males, 53944 ambiguous; 53944 founders) loaded from\n",
      "./data/geno_data/AFR.fam.\n",
      "824452 variants loaded from ./data/geno_data/AFR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 824452 variants remaining.\n",
      "Calculating allele frequencies... 1523313947556371798795done.\n",
      "824452 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 3325 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/DBP_AFR_MIXPRS.txt were skipped due to\n",
      "missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 824413 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/DBP_AFR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 17:12:01 2025\n",
      "PLINK v2.00a6LM 64-bit Intel (6 Aug 2024)      www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2024 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ./result/score/diff_cohort/MIXPRS/PLT_AFR_MIXPRS.log.\n",
      "Options in effect:\n",
      "  --bfile ./data/geno_data/AFR\n",
      "  --extract ./data/snplist_data/AFR_inter_snplist_aou.txt\n",
      "  --out ./result/score/diff_cohort/MIXPRS/PLT_AFR_MIXPRS\n",
      "  --score ./data/final_weight/diff_cohort/MIXPRS/PLT_AFR_MIXPRS.txt header-read\n",
      "\n",
      "Start time: Tue Mar 25 17:12:01 2025\n",
      "15000 MiB RAM detected, ~7642 available; reserving 7500 MiB for main workspace.\n",
      "Using up to 4 compute threads.\n",
      "53944 samples (0 females, 0 males, 53944 ambiguous; 53944 founders) loaded from\n",
      "./data/geno_data/AFR.fam.\n",
      "824452 variants loaded from ./data/geno_data/AFR.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 824452 variants remaining.\n",
      "Calculating allele frequencies... 1523313947556371798795done.\n",
      "824452 variants remaining after main filters.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: --score: 3325 entries in\n",
      "./data/final_weight/diff_cohort/MIXPRS/PLT_AFR_MIXPRS.txt were skipped due to\n",
      "missing variant IDs.\n",
      "(Add the 'list-variants' modifier to see which variants were actually used for\n",
      "scoring.)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--score: 824413 variants processed.\n",
      "--score: Results written to\n",
      "./result/score/diff_cohort/MIXPRS/PLT_AFR_MIXPRS.sscore .\n",
      "End time: Tue Mar 25 17:31:38 2025\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in Height BMI SBP DBP PLT; do\n",
    "\n",
    "for pop2 in AFR; do\n",
    "\n",
    "## MIXPRS\n",
    "method=MIXPRS\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${pop2}_${method}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${pop2}_${method}.txt header-read \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${pop2}_${method}\n",
    "fi\n",
    "\n",
    "done\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "## JointPRS\n",
    "for trait in Height BMI SBP DBP PLT; do\n",
    "for pop2 in AFR; do\n",
    "\n",
    "for pop in EUR EAS AFR; do\n",
    "\n",
    "method=JointPRS\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/MIXPRS/${trait}_${method}_EUR_EAS_AFR_beta_${pop}_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/JointPRS/${trait}_${method}_meta_EUR_EAS_AFR_beta_${pop}.txt \\\n",
    "    --out ./result/score/diff_cohort/MIXPRS/${trait}_${method}_EUR_EAS_AFR_beta_${pop}_prs_${pop2}\n",
    "fi\n",
    "\n",
    "done\n",
    "done\n",
    "\n",
    "done\n",
    "\n",
    "## SDPRX\n",
    "for trait in Height BMI SBP DBP PLT; do\n",
    "for pop2 in AFR; do\n",
    "\n",
    "for pop in EAS AFR; do\n",
    "\n",
    "\n",
    "method=SDPRX\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/MIXPRS/${trait}_${method}_EUR_${pop}_beta_${pop}_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_${pop}_beta_${pop}.txt \\\n",
    "    --out ./result/score/diff_cohort/MIXPRS/${trait}_${method}_EUR_${pop}_beta_${pop}_prs_${pop2}\n",
    "fi\n",
    "\n",
    "done\n",
    "\n",
    "method=SDPRX\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/MIXPRS/${trait}_${method}_EUR_${pop2}_beta_EUR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_beta_EUR.txt \\\n",
    "    --out ./result/score/diff_cohort/MIXPRS/${trait}_${method}_EUR_${pop2}_beta_EUR_prs_${pop2}\n",
    "fi\n",
    "\n",
    "done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in Height BMI SBP DBP PLT; do\n",
    "\n",
    "pop2=AFR\n",
    "\n",
    "## JointPRS_meta\n",
    "method=JointPRS_meta\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/JointPRS/${trait}_${method}_EUR_EAS_AFR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/JointPRS/${trait}_${method}_EUR_EAS_AFR_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/JointPRS/${trait}_${method}_EUR_EAS_AFR_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in Height BMI SBP DBP PLT; do\n",
    "\n",
    "pop2=AFR\n",
    "\n",
    "## JointPRS_linear\n",
    "method=JointPRS_linear\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/JointPRS/${trait}_${method}_EUR_EAS_AFR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/JointPRS/${trait}_${method}_EUR_EAS_AFR_beta_${pop2}.txt header-read \\\n",
    "    --score-col-nums 3 4 5 \\\n",
    "    --out ./result/score/diff_cohort/JointPRS/${trait}_${method}_EUR_EAS_AFR_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in Height BMI SBP DBP PLT; do\n",
    "\n",
    "pop2=AFR\n",
    "\n",
    "## SDPRX\n",
    "method=SDPRX\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in Height BMI SBP DBP PLT; do\n",
    "\n",
    "pop2=AFR\n",
    "\n",
    "## XPASS\n",
    "method=XPASS\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in Height BMI SBP DBP PLT; do\n",
    "\n",
    "pop2=AFR\n",
    "\n",
    "## PRScsx\n",
    "method=PRScsx\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_beta_${pop2}.txt header-read \\\n",
    "    --score-col-nums 3 4 5 \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in Height BMI SBP DBP PLT; do\n",
    "\n",
    "pop2=AFR\n",
    "\n",
    "## MUSSEL\n",
    "method=MUSSEL\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in Height BMI SBP DBP PLT; do\n",
    "\n",
    "pop2=AFR\n",
    "\n",
    "## PROSPER\n",
    "method=PROSPER\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_EAS_AFR_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in Height BMI SBP DBP PLT; do\n",
    "\n",
    "pop2=AFR\n",
    "\n",
    "## PROSPER_update\n",
    "method=PROSPER\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_update_EUR_EAS_AFR_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_update_EUR_EAS_AFR_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_update_EUR_EAS_AFR_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "\n",
    "for trait in Height BMI SBP DBP PLT; do\n",
    "\n",
    "pop2=AFR\n",
    "\n",
    "## BridgePRS\n",
    "method=BridgePRS\n",
    "\n",
    "if [[ ! -e ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_prs_${pop2}.sscore ]]; then\n",
    "plink2 --bfile ./data/geno_data/${pop2} \\\n",
    "    --extract ./data/snplist_data/${pop2}_inter_snplist_aou.txt \\\n",
    "    --score ./data/final_weight/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_beta_${pop2}.txt \\\n",
    "    --out ./result/score/diff_cohort/${method}/${trait}_${method}_EUR_${pop2}_prs_${pop2} \n",
    "fi\n",
    "\n",
    "done"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
